crates/http/src/routes.rs

```diff
@@ -52,50 +69,70 @@ impl Router {
             if let Some(replaced) = replaced {
                 duplicates.push(DuplicateRoute {
-                    route: route.clone(),
-                    replaced_id: replaced,
-                    effective_id: component_id.clone(),
+                    route: replaced.based_route,
+                    replaced_id: replaced.component_id.to_string(),
+                    effective_id,
                 });
             }
         }
 
-        Ok((Self { routes }, duplicates))
-    }
+        // Build a `routefinder` from the remaining routes.
 
-    /// Returns the constructed routes.
-    pub fn routes(&self) -> impl Iterator<Item = (&RoutePattern, &String)> {
-        self.routes.iter()
-    }
+        let mut rf = routefinder::Router::new();
 
-    /// This returns the component id and route pattern for a matched route.
-    pub fn route_full(&self, p: &str) -> Result<(&str, &RoutePattern)> {
-        let matches = self.routes.iter().filter(|(rp, _)| rp.matches(p));
+        for re in routes.into_values() {
+            let (rfroute, parsed) = Self::parse_route(&re.based_route).map_err(|e| {
+                anyhow!(
+                    "Error parsing route {} associated with component {}: {e}",
+                    re.based_route,
+                    re.component_id
+                )
+            })?;
 
-        let mut best_match: (Option<&str>, Option<&RoutePattern>, usize) = (None, None, 0); // matched id, pattern and length
+            let handler = RouteHandler {
+                component_id: re.component_id.to_string(),
+                based_route: re.based_route,
+                raw_route: re.raw_route.to_string(),
+                parsed_based_route: parsed,
+            };
 
-        for (rp, id) in matches {
-            match rp {
-                RoutePattern::Exact(_m) => {
-                    // Exact matching routes take precedence over wildcard matches.
-                    return Ok((id, rp));
-                }
-                RoutePattern::Wildcard(m) => {
-                    // Check and find longest matching prefix of wildcard pattern.
-                    let (_id_opt, _rp_opt, len) = best_match;
-                    if m.len() >= len {
-                        best_match = (Some(id), Some(rp), m.len());
-                    }
-                }
-            }
+            rf.add(rfroute, handler).map_err(|e| anyhow!("{e}"))?;
+        }
+
+        let router = Self {
+            router: std::sync::Arc::new(rf),
+        };
+
+        Ok((router, duplicates))
+    }
+
+    fn parse_route(based_route: &str) -> Result<(routefinder::RouteSpec, ParsedRoute), String> {
+        if let Some(wild_suffixed) = based_route.strip_suffix("/...") {
+            let rs = format!("{wild_suffixed}/*").try_into()?;
+            let parsed = ParsedRoute::trailing_wildcard(wild_suffixed);
+            Ok((rs, parsed))
+        } else if let Some(wild_suffixed) = based_route.strip_suffix("/*") {
```

Comment:
If i'm understanding this correctly, wildcard routes can now either be specified with `...` or `*`, is that right? 

If that's the case (It's not a hill i'm willing to die on) but i'd vote for supporting just `...`.

Reply:
Yeah I don't have a strong view on this.  The sorta kinda reasoning was that if people are familiar with `routefinder` style strings, then we shouldn't make them memorise a seemingly capricious difference.  In particular, the Rust SDK `Router` uses unmodified `routefinder` strings (I believe), and it felt weird to largely accept the same strings but with an "except you have to use `/...` here and `/*` there" clause.  But I can validate this away for sure.

Reply:
That sounds reasonable to me!